### PR TITLE
Fixed typos in comments for $watch() methods

### LINKED ISSUE #599: Incorrect event names in API doc for $firebaseArray.$watch
The API doc for ```$firebaseArray.$watch``` (https://github.com/firebase/angularfire/blob/master/src/FirebaseArray.js#L245) is incorrect.

Wrong: ```<code>{ type: 'added|updated|moved|removed', key: 'key_of_item_affected'}</code>```
Right: ```<code>{ event: 'child_added|child_updated|child_moved|child_removed', key: 'key_of_item_affected'}</code>```

### PR DESCRIPTION
**Pull request by jwngr**
Fixes #599

### CODE REVIEW

**Comment by katowulf** [comment]
I'm of a mind to remove the events entirely. Our behavior isn't consistent with Angular's $watch (which just triggers any time there is a change) and will encourage people attempting to use the binding services as a substitute for the more appropriate SDK. This would be a breaking change, but in my opinion, the inclusion of these arguments is both incorrect and against the spirit of the method.

File: src/FirebaseObject.js [hunk 1]
```diff
@@ -151,7 +151,7 @@
         /**
          * Listeners passed into this method are notified whenever a new change is received
          * from the server. Each invocation is sent an object containing
-         * <code>{ type: 'updated', key: 'my_firebase_id' }</code>
+         * <code>{ type: 'child_updated', key: 'my_firebase_id' }</code>
```
Comment (katowulf) [inline src/FirebaseObject.js]: this should be `value` https://github.com/firebase/angularfire/blob/jw-watch-docs/src/FirebaseObject.js#L244

Comment (jwngr) [inline src/FirebaseObject.js]: Umm, yeah you're right...

**Comment by jwngr** [comment]
I can't say I disagree with you. We will have to wait until we bump to v2.0.0 to do this. Let's get the comments right in the meantime.

**Comment by jwngr** [comment]
@katowulf - ping

**Pull request merged**